### PR TITLE
Fixup build numbers

### PR DESCRIPTION
**Pull request by hmaarrfk**
We should check the output of the recipes, and merge if we see nompi with build number 105

Build 4 didn't generate build numbers of 4 and 104, just both of build 4.

https://github.com/conda-forge/fftw-feedstock/pull/85#issuecomment-1233611936
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,29 +30,14 @@ source:
...
-      # `pkg * mpi_mpich_*` for mpich
-      # `pkg * mpi_*` for any mpi
-      # `pkg * nompi_*` for no mpi
-      {% if mpi != 'nompi' %}
-      {% set mpi_prefix = "mpi_" + mpi %}
-      {% else %}
-      {% set mpi_prefix = "nompi" %}
-      {% endif %}
-      string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}"
+      string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:40]: Truthfully i don't really know why this change of syntax changed things. But the build logs show that it was effective.

**Comment by hmaarrfk** [comment]
I want to see if i can get the tests to pass on travis too so I'll wait until the builds fail to try to either move things to azure, or to disable the tests on travis.

Right now, 100% of the builds failed on traivis due to time.

**Comment by egpbos** [comment]
Do you mean 100% of conda-forge Travis tests? Or just fftw-feedstock? I think the build fails differently now, before it failed after 50 minutes, right?

**Comment by hmaarrfk** [comment]
Travis tests only.

This job is failing at 50 min. https://app.travis-ci.com/github/conda-forge/fftw-feedstock/jobs/581516002

Other jobs have been the same.

**Comment by egpbos** [comment]
Right, but the latest Travis run in this PR failed after 6m. Is there something different going on that needs to be fixed?

**Comment by hmaarrfk** [comment]
no, i think travis a bit flaky sometimes.

The one you merged before was also problematic.

Lets see if cross compilation works.

**Comment by hmaarrfk** [comment]
this is ready for review

**Comment by egpbos** [comment]
@jayfurmanek, since you added them according to git blame, do you have any ideas about how to fix the ppc tests here?

**Comment by egpbos** [comment]
Let's just merge this for now so that the build numbers are fixed and fix the PPC tests in another PR.

**Comment by egpbos** [comment]
And thanks again @hmaarrfk!
